### PR TITLE
'Headline-only' endpoint for emails

### PR DESCRIPTION
**Pull request by philmcmahon**
## What does this change?
This modifies the facia email endpoint so that if the ?format=email-headline parameter is given then only the headline will be returned. 

## What is the value of this and can you measure success?
Better email titles in exact target!

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: facia/app/controllers/FaciaController.scala [hunk 1]
```diff
@@ -111,27 +114,25 @@ trait FaciaController extends BaseController with Logging with ImplicitControlle
   private[controllers] def renderFrontPressResult(path: String)(implicit request: RequestHeader) = {
     val futureResult = frontJsonFapi.getLite(path).flatMap {
       case Some(faciaPage) =>
-        successful(
+        successful(Cached(CacheTime.Facia)(
```
Comment (QuarpT) [inline facia/app/controllers/FaciaController.scala]: do we really want to cache all request types?

File: facia/app/controllers/FaciaController.scala [hunk 2]
```diff
@@ -111,27 +114,25 @@ trait FaciaController extends BaseController with Logging with ImplicitControlle
...
+            RevalidatableResult(Ok(body).as("text/xml; charset=utf-8"), body)
           }
           else if (request.isJson)
-            Cached(CacheTime.Facia)(JsonFront(faciaPage))
+            JsonFront(faciaPage)
           else if (request.isEmail || ConfigAgent.isEmailFront(path)) {
-            val htmlResponse = FrontEmailHtmlPage.html(faciaPage)
-            Cached(CacheTime.Facia) {
+            if (shouldOnlyReturnHeadline(request)) {
+                RevalidatableResult.Ok(faciaPage.collections.head.curated.head.header.headline)
```
Comment (QuarpT) [inline facia/app/controllers/FaciaController.scala]: I'm against list.head but it's a subjective opinion 😄

**Review by QuarpT — APPROVED** [review]
Looks good. Funny that we have to parse megabytes of pressed json to get 1 headline. I guess this endpoint will hardly be used?

**Comment by philmcmahon** [comment]
Thanks for the comments! Yeah this should only be used by exact target.

File: facia/app/controllers/FaciaController.scala [hunk 3]
```diff
@@ -123,7 +123,11 @@ trait FaciaController extends BaseController with Logging with ImplicitControlle
             JsonFront(faciaPage)
           else if (request.isEmail || ConfigAgent.isEmailFront(path)) {
             if (shouldOnlyReturnHeadline(request)) {
-                RevalidatableResult.Ok(faciaPage.collections.head.curated.head.header.headline)
+              val headline = for {
+                topCollection <- faciaPage.collections.headOption
+                topCurated <- topCollection.curated.headOption
+              } yield topCurated.header.headline
+                RevalidatableResult.Ok(headline.getOrElse("Error: Could not extract headline from front"))
```
Comment (QuarpT) [inline facia/app/controllers/FaciaController.scala]: The error shouldn't be served as a 200 response

**Pull request merged**